feature-extensions/multiaccount/triggerJenkinsDeleteResources.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import argparse
+import subprocess
+import requests
+import time
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--jenkins-url', help='Specify the Jenkins url')
+    parser.add_argument('--jenkins-username',
+                        help='Specify the Jenkins username')
+    parser.add_argument('--jenkins-password',
+                        help='Specify the Jenkins password')
+    parser.add_argument('--account-details',
+                        help='Specify the Accounts to delete')
+    args = parser.parse_args()
+    startJob(args)
+
+
+def startJob(args):
+    collectUserInput(args)
+    subprocess.check_call(
+        [
+            "curl",
+            "-sL",
+            ("http://%s/jnlpJars/jenkins-cli.jar") %
+            (args.jenkins_url),
+            "-o",
+            "jenkins-cli.jar"])
+    jenkins_build_command = \
+        ' java -jar jenkins-cli.jar -s http://%s '\
+        '-auth %s:%s build delete-resources  -p input=%s' \
+        % (args.jenkins_url, args.jenkins_username,
+            args.jenkins_password, args.account_details)
+
+    subprocess.call(jenkins_build_command, shell=True)
+    while True:
+        data = jenkins_job_status("delete-resources", args)
+        if data['result'] == "SUCCESS":
+            print 'Job run Successfully'
+            break
+        elif data['building']:
+            print 'Job is Running'
+            time.sleep(60)
```

Comment:
Please do not use hard, fixed `sleep`s in production code. It's bad practice generally and always a worst-fit solution, even if the number picked happens to be the right guess for the particular hardware and environment the code is running on.

Use a library we already pull in (Retrying) to do multiple bounded retries at shorter intervals.

Adding in minute-long waits everywhere in the installer and feature packs tends to unnecessarily balloon the amount of time it takes for the installer to run, and fixed wait times are a guess based on the current environment and introduce instability and nondeterminism when the codebase is run in different environments.

Reply:
As mentioned, updated with Retrying Python Package.